vendor/github.com/go-jose/go-jose/v4/jwe.go

```diff
@@ -288,10 +288,11 @@ func ParseEncryptedCompact(
 	keyAlgorithms []KeyAlgorithm,
 	contentEncryption []ContentEncryption,
 ) (*JSONWebEncryption, error) {
-	parts := strings.Split(input, ".")
-	if len(parts) != 5 {
+	// Five parts is four separators
+	if strings.Count(input, ".") != 4 {
 		return nil, fmt.Errorf("go-jose/go-jose: compact JWE format must have five parts")
 	}
+	parts := strings.SplitN(input, ".", 5)
```

Comment:
Changes look minimal, low-risk, although I'm not sure why they didn't use the `SplitN` result to get the number of components, but setting the limit to `expected + 1` (and produce an error if not the expected number)